app/controllers/candidate_interface/application_form_controller.rb

```diff
@@ -82,5 +93,11 @@ def further_information_params
       )
         .transform_values(&:strip)
     end
+
+    def build_previous_application_form
+      return if params['candidate_id']&.to_i != current_candidate.id
+
+      @application_form = current_candidate.application_forms.find(params['id']) if params['id'].present?
```

Comment:
I tried to say before in https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/pull/2016#discussion_r420847452 that it's impossible for `params['id']` to be nil. That's because the user can only reach this code by visiting the `/application/:id/submitted` route, so the `params[:id]` must always exist.

Reply:
Yea i was having a struggle day yesterday haha. Thanks, i get it now.